src/server/conn/auto.rs

```diff
@@ -101,6 +105,26 @@ impl<E> Builder<E> {
         Http2Builder { inner: self }
     }
 
+    /// Only accepts HTTP/2
+    ///
+    /// Does not do anything if used with [`serve_connection_with_upgrades`]
+    #[cfg(feature = "http2")]
+    pub fn http2_only(mut self) -> Self {
```

Comment:
I'd instead suggest:

```rust
    pub fn http2_version(mut self, v: Option<Version>) -> Self
```

... and expose `Version` struct. Otherwise, it's panicable when you call with `http1_only = true and http2_only = true`, or `self.http1_only().http2_only()`. It's hard to tell if a passed builder is already configured. `version` setter override the configured value, at least better than assert! not set.